web/src/components/layout/layout.tsx

```diff
@@ -180,9 +200,16 @@ class Layout extends React.PureComponent<LayoutProps, LayoutState> {
       'staging-banner-is-visible': showStagingBanner,
     });
 
+    const alreadyEnrolled =
+      this.state.showWelcomeModal &&
+      user.account !== null &&
+      user.account.enrollment.challenge !== null;
```

Comment:
If `enrollment` is undefined this will throw an exception. Does this work?

```
const alreadyEnrolled = this.state.showWelcomeModal && user.account && user.account.enrollment;
```

I don't think it's possible for `enrollment` to exist as an empty object, but you could also check `user.account.enrollment.challenge` for extra safety if you wanna.

Reply:
Testing on a brand new user locally, I think the `enrollment` property gets automatically created on the account per lines 179-182 of `auth-router.ts`. 

For sake of clean cascade though, decided to go with: 

```this.state.showWelcomeModal &&
      user.account &&
      user.account.enrollment &&
      user.account.enrollment.challenge;```